pkg/certificate/mrc_reconciler.go

```
@@ -1,214 +1,176 @@
-			mrc.Status.ComponentStatuses.MutatingWebhook = status
-		case UseCaseXDSControlPlane:
-			mrc.Status.ComponentStatuses.XDSControlPlane = status
-		case UseCaseSidecar:
-			mrc.Status.ComponentStatuses.Sidecar = status
-		case UseCaseBootstrap:
-			mrc.Status.ComponentStatuses.Bootstrap = status
-		case UseCaseGateway:
-			mrc.Status.ComponentStatuses.Gateway = status
-		}
+	// no update required if the issuers are already set to the desired value
+	if signingIssuer.ID == desiredSigningMRC.Name && validatingIssuer.ID == desiredValidatingMRC.Name {
+		log.Debug().Msgf("issuers already set to the desired value. Will not update issuers: validating[%s] and signing[%s]", validatingIssuer.ID, signingIssuer.ID)
+		return false, nil
 	}
 
-	return m.mrcClient.UpdateMeshRootCertificate(mrc)
+	bothActive := desiredSigningMRC.Spec.Intent == v1alpha2.ActiveIntent && desiredValidatingMRC.Spec.Intent == v1alpha2.ActiveIntent
+	exist := (desiredSigningMRC.Name == signingIssuer.ID || desiredSigningMRC.Name == validatingIssuer.ID) &&
+		(desiredValidatingMRC.Name == signingIssuer.ID || desiredValidatingMRC.Name == validatingIssuer.ID)
+
+	// if desiredSigningMRC != desiredValidatingMRC and both MRCs have active intents, their state is non
+	// deterministic. No update required if the current signing and validating issuers correspond to the
+	// existing MRCs. This check is necessary to avoid continuously resetting the issuers on start up
+	if desiredSigningMRC != desiredValidatingMRC && bothActive && exist {
+		log.Debug().Msgf("Will not update issuers to avoid repeated updates; validating[%s] and signing[%s]", validatingIssuer.ID, signingIssuer.ID)
+		return false, nil
+	}
+
+	return true, nil
 }
 
-// isTerminal checks if the MRC is in a terminal state.
-func isTerminal(mrc *v1alpha2.MeshRootCertificate) bool {
-	// TODO(5046): check if the mrc is in a terminal state.
-	return true
+func (m *Manager) updateIssuers(signingIssuer, validatingIssuer *issuer) error {
```

this function doesn't need to return an error